api/grpc/mpi/v1/file_stream.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+package v1
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"google.golang.org/grpc"
+)
+
+var (
+	ErrInvalidHeader     = errors.New("invalid header")
+	ErrUnexpectedContent = errors.New("unexpected content")
+	ErrSend              = errors.New("send error")
+	ErrWrite             = errors.New("write error")
+	ErrFailedRead        = errors.New("failed to read")
+)
```

Comment:
nit: package vars are mutable, and the compiler will happily allow moronic code like `v1.ErrInvalidHeader = errors.New("haha, I just broke all your error handling")`

I'm not sure if this repo has conventions around errors, but we can get compiler guarantees with a little more code.

```suggestion

// StreamError is an error related to streaming files. Use the constants for more specific causes.
type StreamError string

func (s StreamError) Error() string { return string(s) }

const (
	ErrInvalidHeader     StreamError = "invalid header"
	ErrUnexpectedContent StreamError = "unexpected content"
	ErrSend              StreamError = "send error"
	ErrWrite             StreamError = "write error"
	ErrFailedRead        StreamError = "failed to read"
)
```

Reply:
TBH there's a lot of bikeshedding possible with golang errors. For N4A we kinda settled on:

- if callers want to handle different kinds of errors, define an error type with a handful of methods to differentiate the kind of error (kinda like [`url.Error`](https://pkg.go.dev/net/url@go1.24.1#Error)) and return that. Often with `Is*Err(error) bool` funcs to make it easier on callers.
- else, use `fmt.Errorf("<what failed>: %w", err)` or `errors.New("<what failed>")`

`fmt.Errorf` sounds good to me here; can always make it more complicated later when there's an actual caller.

Reply:
removed, and use `fmt.Errorf`